tests/api_jobs/test_base.py

```diff
@@ -91,7 +91,8 @@ def test_ApiJob_other_error(mocker):
     mock_api_client = mocker.MagicMock()
     mock_session = mocker.MagicMock()
 
-    api_job._do_call_api(mock_api_client, mock_session)
+    with pytest.raises(Exception):
```

Comment:
A failed job should always raise an exception so the queue processor can log and continue, pause the queue, or drop the job.